### ISSUE TITLE
[C++] [Acero] why merging multiple SwissTable into one before probing

### ISSUE THREAD

**Issue by XuHuaiyu** [apache/arrow#39650]
### Describe the usage question you have. Please include as many useful details as  possible.

While reading the Acero SwissJoin source code, I have a question: why are Swiss tables for different partitions constructed in parallel, and then physically merged into a single SwissTable?
Can we not perform probe operations on multiple SwissTables?

### Component(s)

C++
